app/code/community/Reverb/ReverbSync/Model/Observer.php

```diff
@@ -7,11 +7,24 @@ class Reverb_ReverbSync_Model_Observer
     //function to create the product in reverb
     public function productSave($observer)
     {
-        $productSyncHelper = Mage::helper('ReverbSync/sync_product');
-        $product_id = $observer->getProduct()->getId();
-        try
-        {
-            $listingWrapper = $productSyncHelper->executeIndividualProductDataSync($product_id);
+      try
+      {
+          $product = $observer->getProduct();
+          $product_id = $product->getId();
+
+          $syncToReverb = $product->getData('rev_sync');
+          if (is_null($syncToReverb)) {
+              // TODO: This will potentially generate one SQL query per product saved.  Revamp this to queue the check and process in async batches.
+              $syncToReverb = Mage::getResourceModel('catalog/product')->getAttributeRawValue($product_id, 'rev_sync');
```

Comment:
There's a couple things here. The first is that this event will fire during any product->save() call, which can be done on partially loaded products. The $product->getData('rev_sync') call is way to see if the attribute has been loaded on the product or not.

The other part of it is that the extension is currently passing around the Product ID rather than the loaded Product instance. You can see that this event handler calls:

$listingWrapper = $productSyncHelper->executeIndividualProductDataSync($product_id);

Which will only pass the product ID, requiring the product to be loaded again (regardless of whether or not it already has been).

Fixing that is likely larger than I can tackle for MSL, but I'd suggest trying to keep .load() calls to a minimum and working with collections rather than individual instances--the gains should be significant.